packages/graph/src/graph2/layouts/dagre.ts

```diff
@@ -59,10 +65,12 @@ export class Dagre extends Layout {
                 });
                 response.links.forEach(l => {
                     const e = data.edge(l.id);
+                    const sourceDist = distance(e.source.x, e.source.y, l.points[0][0] + size.width / 2, l.points[0][1] + size.height / 2);
+                    const targetDist = distance(e.target.x, e.target.y, l.points[0][0] + size.width / 2, l.points[0][1] + size.height / 2);
                     e.points = [
-                        [e.source.x, e.source.y],
+                        sourceDist < targetDist ? [e.source.x, e.source.y] : [e.target.x, e.target.y],
                         ...l.points.map(p => [p[0] + size.width / 2, p[1] + size.height / 2]),
-                        [e.target.x, e.target.y]
+                        sourceDist < targetDist ? [e.target.x, e.target.y] : [e.source.x, e.source.y]
```

Comment:
What's happening here? If the source position is farther from the center than the target position then swap the edge starting and ending points? Wouldn't we also need to reverse the order of the points inbetween?